### DISCUSSION TITLE
Odd characters added when attempting to color text on input field

### DISCUSSION THREAD

**Discussion by weagle08** [Hopding/pdf-lib#1042] [Q&A] [answered]
Creating discussion instead of an issue because I'm sure this is me misunderstanding something. I'm attempting to color text on an input field and there are extra characters being added at the beginning. I'm doing the following:

```
    let firstNameField = form.getTextField('topmostSubform[0].Page1[0].First_Name_Given_Name[0]');
    firstNameField.acroField.getWidgets().forEach((w) => {
        const { context } = w.dict;
        const { width, height } = w.getRectangle();

        let hexText = PDFHexString.fromText('My Name');

        const appearance = [               
            ...drawText(hexText, {
                x: 0,
                y: 0,
                color: rgb(1, 0, 0),
                rotate: degrees(0),
                xSkew: degrees(0),
                ySkew: degrees(0),
                font: StandardFonts.Helvetica,
                size: 12
            })
        ];

        const stream = context.formXObject(appearance, {
            BBox: context.obj([0, 0, width, height]),
            Matrix: context.obj([1, 0, 0, 1, 0, 0])
        });

        const streamRef = context.register(stream);
        w.setNormalAppearance(streamRef);
    });
```

I got this sample code from @Hopding  [sample here](https://github.com/Hopding/pdf-lib/issues/418#issuecomment-697052730). It works, as in my text is inserted into the field and it is colored red, but there are a couple extra characters.

![InvalidCharacters](https://user-images.githubusercontent.com/7888478/137940061-015e52df-5cca-4e47-b459-dffa0c25f742.png)

**Comment by weagle08** [comment]
Here is a runnable project that will reproduce the issue.

1. extract
2. npm install
3. npm run build
4. node dist/Pdf

If you use VS Code you can open and run with debugger.

[pdf-test.zip](https://github.com/Hopding/pdf-lib/files/7374811/pdf-test.zip)

**Comment by weagle08** [comment, marked as answer]
So the issue seems to be in the ```PDFHexString.fromText()``` method. It pads the start of the text and this is the source of the problem... I implemented a simple toHex function and all works as expected. Is there an issue with the PDFHexString.fromText() method that needs to be filed as a bug or am I using this method for the wrong purpose?

```
    toHexString(str: string) {
        let retVal = '';
        for (let i = 0; i < str.length; i++) {
            let hex = str.charCodeAt(i).toString(16);
            retVal += hex;
        }

        return retVal;
    }
```

**Reply by Hopding** [reply]
@weagle08 looks like you should be using the [`PDFFont.encodeText`](https://pdf-lib.js.org/docs/api/classes/pdffont#encodetext) method.
